ethcore/src/engines/clique/step_service.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright 2015-2019 Parity Technologies (UK) Ltd.
+// This file is part of Parity Ethereum.
+
+// Parity Ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Ethereum.  If not, see <http://www.gnu.org/licenses/>.
+
+use engines::Engine;
+use parity_machine::Machine;
+use std::sync::Weak;
+use std::time::Duration;
+use std::thread;
+use std::sync::Arc;
+use parking_lot::RwLock;
+
+pub struct StepService {
+	shutdown: Arc<RwLock<bool>>,
+	thread: Option<thread::JoinHandle<()>>,
+}
+
+impl StepService {
+	pub fn start<M: Machine + 'static>(engine: Weak<Engine<M>>) -> Arc<Self> {
+		let shutdown = Arc::new(RwLock::new(false));
```

Comment:
I don't understand the point of `shutdown` and `shutdown1`

Reply:
it is to move one `Arc` into the thread and return the other perhaps the naming could be improved!